File: striver/S1L4/python/reverse_number.py
def reverse_number(num):
    """
    Reverse a given number
    Example: 1234 -> 4321
    """
    reversed_num = 0
    while num > 0:
        digit = num % 10
        reversed_num = reversed_num * 10 + digit
        num //= 10
    return reversed_num

if __name__ == "__main__":
    num = int(input("Enter a number: "))
    print(f"Reverse of {num} is: {reverse_number(num)}") 

File: striver/S1L4/java/ReverseNumber.java
import java.util.Scanner;

public class ReverseNumber {
    public static int reverseNumber(int num) {
        int reversedNum = 0;
        while (num > 0) {
            int digit = num % 10;
            reversedNum = reversedNum * 10 + digit;
            num /= 10;
        }
        return reversedNum;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();
        System.out.println("Reverse of " + num + " is: " + reverseNumber(num));
        sc.close();
    }
} 